File: src/main/java/cloud/api/RebalanceDataBlocks.java
package cloud.api;

import cloud.common.NodeConfiguration;
import se.sics.kompics.Event;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-04-21
 *
 */

public class RebalanceDataBlocks extends Event {

	private NodeConfiguration nodeConfiguration;

	public RebalanceDataBlocks(NodeConfiguration nodeConfiguration) {
		this.nodeConfiguration = nodeConfiguration;
	}

	public NodeConfiguration getNodeConfiguration() {
		return nodeConfiguration;
	}

}

File: src/main/java/cloud/api/RebalanceDataBlocks.scala
package cloud.api

import se.sics.kompics.Event
import cloud.common.NodeConfiguration
import reflect.BeanProperty

/**
 * @author Amir Moulavi
 * @date 2011-07-16
 *
 */
class RebalanceDataBlocks(@BeanProperty var nodeConfiguration:NodeConfiguration) extends Event